linera-core/src/updater.rs

```diff
@@ -279,10 +279,22 @@ where
 
         // Find the missing blobs locally and retry.
         let required = match certificate.value() {
+            #[cfg(not(with_testing))]
```

Comment:
Let's not have test-only code here.
Instead, when executing a `ReadBlob` operation for a block we're missing, we just fail execution at that point.

Reply:
Ah, I see!
Let's remove the check that all blobs submitted with a block _proposal_ are actually required. We aren't supposed to store them at this point anyway. (https://github.com/linera-io/linera-protocol/issues/2199)

Reply:
So this is only called either for confirmed certificates of when we're trying to finalize a block, so this won't be called for block proposals 🤔
For block proposals if we don't find a blob when calling `handle_block_proposal` in the validators, we just return the error. I could just remove the required check entirely 🤔
I think that should be fine because we only get this error from `check_no_missing_blobs`, which gets the required blobs to check there from the oracle outcomes. So we shouldn't have to check again here 🤔